app/assets/javascripts/discourse/app/components/d-navigation.js

```diff
@@ -11,6 +11,20 @@ export default Component.extend(FilterModeMixin, {
   router: service(),
   dialog: service(),
   tagName: "",
+  canShowTopicButton: false,
+
+  didInsertElement() {
+    this.set("canShowTopicButton", this.canCreateTopic);
```

Comment:
why is this relaying on this.canCreateTopic rather than canShowTopicAction()?

Reply:
@Ledespana this.canCreateTopic is how that used to be determined by default

Reply:
do you need the didInsertElement? does this.canCreateTopic rely on something in the DOM?
Can't you in line 14 do the following and avoid using didInsertElement?
```
...
canShowTopicButton:  this.canCreateTopic,
```

Reply:
@Ledespana will try this

Reply:
@Ledespana addresses using an earlier suggestion